### PR TITLE
remove wg-quick calls from freebsd

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

File: wireguard/wireguard_freebsd.go [hunk 1]
```diff
@@ -1,98 +1,157 @@
...
 }
 
-func (nc *NCIface) getKernelLink() *netLink {
-	linkAttrs := netlink.NewLinkAttrs()
-	linkAttrs.Name = nc.Name
-	return &netLink{
-		attrs: &linkAttrs,
-	}
+func apply(n *config.Node, c *wgtypes.Config) error {
+	return nil
```
Comment (abhishek9686) [inline wireguard/wireguard_freebsd.go:156]: looks like this func is not doing any operations, can be removed right?

Comment (0xdcarns) [reply wireguard/wireguard_freebsd.go:156]: it is being used to overwrite the operation of the main wireguard.go file

File: wireguard/wireguard_freebsd.go [hunk 2]
```diff
@@ -1,98 +1,157 @@
...
-const kernelModule = "/boot/mdodules/if_wg.ko"
+const kernelModule = "/boot/modules/if_wg.ko"
 
 // Create - creates a linux WG interface based on a node's given config
 func (nc *NCIface) Create() error {
-	if _, err := os.Stat(kernelModule); err == nil {
-		return ApplyWGQuickConf(nc)
+	if _, err := os.Stat(kernelModule); err != nil {
+		logger.Log(3, "using userspace wireguard")
+		return nc.createUserSpaceWG()
```
Comment (0xdcarns) [inline wireguard/wireguard_freebsd.go:23]: in future need to check if tun module installed for userspace build to function

File: wireguard/wireguard_freebsd.go [hunk 3]
```diff
@@ -1,98 +1,157 @@
...
+		return err
+	}
+	if _, err := ncutils.RunCmd(fmt.Sprintf("%s setconf %s %s", wg, nc.Name, os.TempDir()+"/netmaker.conf"), true); err != nil {
+		return err
+	}
+	if err := os.Remove(os.TempDir() + "/netmaker.conf"); err != nil {
+		return err
+	}
+	//add addresses
+	for _, address := range nc.Addresses {
```
Comment (0xdcarns) [inline wireguard/wireguard_freebsd.go:103]: is there a reason this logic isn't in the ApplyAddrs func?

File: wireguard/wireguard_freebsd.go (hunk 2 above)
Comment (mattkasun) [reply wireguard/wireguard_freebsd.go:23]: added to GRA-749

File: wireguard/wireguard_freebsd.go (hunk 3 above)
Comment (mattkasun) [reply wireguard/wireguard_freebsd.go:103]: AppyAddrs is never called in freebsd.

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**